Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -631,7 +660,7 @@ def _data_update(artists, workspace, new_kwargs=None):
             # Therefore we need to autoscale regardless of autoscale_on_update.
             if self.lines:
                 # Otherwise set autoscale to autoscale_on_update.
-                self.set_autoscaley_on(autoscale_on_update)
+                self.set_autoscaley_on(autoscale)
```

Comment:
As discussed offline, the best thing to do here is probably incorporate the `if` statement into the context manger 👍